### PR TITLE
Updated documentation to refer to GeoCAT-ncomp

### PR DESCRIPTION
**Pull request by khallock**
[Temporary Readthedocs](https://geocat-ncomp.readthedocs.io/en/docs_cleanup/) site built from this `docs_cleanup` branch for reference.

### CODE REVIEW

File: docs/user_api/index.rst [hunk 1]
```diff
@@ -4,35 +4,31 @@ User API
...
-   geocat.comp.rcm2points
+   geocat.ncomp.linint2_points
 
-   geocat.comp.rcm2rgrid
+   geocat.ncomp.rgrid2rcm
 
-   geocat.comp.linint2_points
+   geocat.ncomp.triple2grid
 
-   geocat.comp.rgrid2rcm
```
Comment (erogluorhan) [inline docs/user_api/index.rst:34]: Is it intentional not to add `geocat.ncomp.rgrid2rcm` ?

File: docs/user_api/index.rst [hunk 2]
```diff
@@ -4,35 +4,31 @@ User API
...
-
-   geocat.comp.triple2grid
```
Comment (erogluorhan) [inline docs/user_api/index.rst:36]: Is it intentional not to add `geocat.ncomp. triple2grid ` ?

File: docs/user_api/index.rst (hunk 1 above)
Comment (khallock) [reply docs/user_api/index.rst:34]: `geocat.comp.ndpolyfit` and `geocat.comp.ndpolyval` were removed, so everything below those got shifted up by two lines. `rgrid2rcm` is now on line 30, `triple2grid` is on line 32.

File: docs/user_api/index.rst (hunk 2 above)
Comment (khallock) [reply docs/user_api/index.rst:36]: See above

File: docs/user_api/index.rst (hunk 1 above)
Comment (erogluorhan) [reply docs/user_api/index.rst:34]: Got it, thanks.

File: docs/user_api/index.rst (hunk 2 above)
Comment (erogluorhan) [reply docs/user_api/index.rst:36]: Thanks

**Pull request merged**